File: Python/7kyu/_123_Testing.py
import unittest
from VovewlCount import getCount
from Mumbling import accum
from DisemvowelTrolls import disemvowel
from HighestAndLowest import high_and_low
from square_every_digit import square_digits
from Descending_Order import descending_order
from GetTheMiddleCharacter import get_middle
from Isogram import is_isogram, is_isogram1
from ListFiltering import filter_list
from ExesAndOhs import XO
from ValidatePIN import validate_pin


class Testing_7kyu(unittest.TestCase):
    def test_VowelCount(self):
        self.assertEqual(getCount("aeiou"), 5, f'Incorrect answer for "aeiou"')

        self.assertEqual(getCount("y"), 0, f'Incorrect answer for "y"')

        self.assertEqual(
            getCount("bcdfghjklmnpqrstvwxz y"),
            0,
            f'Incorrect answer for "bcdfghjklmnpqrstvwxz y"',
        )
        self.assertEqual(getCount(""), 0, f"Incorrect answer for empty string")

        self.assertEqual(
            getCount("abracadabra"), 5, f'Incorrect answer for "abracadabra"'
        )

    def test_Mumbling(self):
        self.assertEqual(
            accum("ZpglnRxqenU"),
            "Z-Pp-Ggg-Llll-Nnnnn-Rrrrrr-Xxxxxxx-Qqqqqqqq-Eeeeeeeee-Nnnnnnnnnn-Uuuuuuuuuuu",
        )
        self.assertEqual(
            accum("NyffsGeyylB"),
            "N-Yy-Fff-Ffff-Sssss-Gggggg-Eeeeeee-Yyyyyyyy-Yyyyyyyyy-Llllllllll-Bbbbbbbbbbb",
        )
        self.assertEqual(
            accum("MjtkuBovqrU"),
            "M-Jj-Ttt-Kkkk-Uuuuu-Bbbbbb-Ooooooo-Vvvvvvvv-Qqqqqqqqq-Rrrrrrrrrr-Uuuuuuuuuuu",
        )
        self.assertEqual(
            accum("EvidjUnokmM"),
            "E-Vv-Iii-Dddd-Jjjjj-Uuuuuu-Nnnnnnn-Oooooooo-Kkkkkkkkk-Mmmmmmmmmm-Mmmmmmmmmmm",
        )
        self.assertEqual(
            accum("HbideVbxncC"),
            "H-Bb-Iii-Dddd-Eeeee-Vvvvvv-Bbbbbbb-Xxxxxxxx-Nnnnnnnnn-Cccccccccc-Ccccccccccc",
        )

    def test_DisemvowelTrolls(self):
        self.assertEqual(
            disemvowel(
                "This website is for losers LOL!"), "Ths wbst s fr lsrs LL!"
        )

    def test_HighAndLow(self):
        self.assertEqual(
            high_and_low("4 5 29 54 4 0 -214 542 -64 1 -3 6 -6"), "542 -214"
        )
        self.assertEqual(high_and_low("1 -1"), "1 -1")
        self.assertEqual(high_and_low("1 1"), "1 1")
        self.assertEqual(high_and_low("-1 -1"), "-1 -1")
        self.assertEqual(high_and_low("1 -1 0"), "1 -1")
        self.assertEqual(high_and_low("1 1 0"), "1 0")
        self.assertEqual(high_and_low("-1 -1 0"), "0 -1")
        self.assertEqual(high_and_low("42"), "42 42")

    def test_square_digits(self):
        self.assertEqual(square_digits(9119), 811181)
        self.assertEqual(square_digits(0), 0)

    def test_descending_order(self):
        self.assertEqual(descending_order(0), 0)
        self.assertEqual(descending_order(1), 1)
        self.assertEqual(descending_order(111), 111)
        self.assertEqual(descending_order(15), 51)
        self.assertEqual(descending_order(1021), 2110)
        self.assertEqual(descending_order(123456789), 987654321)

    def test_get_the_middle_character(self):
        self.assertEqual(get_middle("test"), "es")
        self.assertEqual(get_middle("testing"), "t")
        self.assertEqual(get_middle("middle"), "dd")
        self.assertEqual(get_middle("A"), "A")
        self.assertEqual(get_middle("of"), "of")

    def test_is_isogram(self):
        self.assertEqual(is_isogram("Dermatoglyphics"), True)
        self.assertEqual(is_isogram("isogram"), True)
        self.assertEqual(is_isogram("aba"), False,
                         "same chars may not be adjacent")
        self.assertEqual(is_isogram("moOse"), False,
                         "same chars may not be same case")
        self.assertEqual(is_isogram("isIsogram"), False)
        self.assertEqual(is_isogram(""), True,
                         "an empty string is a valid isogram")

        self.assertEqual(is_isogram1("Dermatoglyphics"), True)
        self.assertEqual(is_isogram1("isogram"), True)
        self.assertEqual(is_isogram1("aba"), False,
                         "same chars may not be adjacent")
        self.assertEqual(is_isogram1("moOse"), False,
                         "same chars may not be same case")
        self.assertEqual(is_isogram1("isIsogram"), False)
        self.assertEqual(is_isogram1(""), True,
                         "an empty string is a valid isogram")

    def test_list_filtering(self):
        self.assertEqual(filter_list([1, 2, 'a', 'b']), [1, 2])
        self.assertEqual(filter_list([1, 'a', 'b', 0, 15]), [1, 0, 15])
        self.assertEqual(filter_list(
            [1, 2, 'aasf', '1', '123', 123]), [1, 2, 123])

    def test_exes_and_ohs(self):
        self.assertEqual(XO("xo"), True)
        self.assertEqual(XO("xo0"), True)
        self.assertEqual(XO("xxxxxo"), False)

    def test_validate_pin(self):
        self.assertEqual(validate_pin("1"), False, "Wrong output for '1'")
        self.assertEqual(validate_pin("12"), False, "Wrong output for '12'")
        self.assertEqual(validate_pin("123"), False, "Wrong output for '123'")
        self.assertEqual(validate_pin("12345"), False,
                         "Wrong output for '12345'")
        self.assertEqual(validate_pin("1234567"), False,
                         "Wrong output for '1234567'")
        self.assertEqual(validate_pin("-1234"), False,
                         "Wrong output for '-1234'")
        self.assertEqual(validate_pin("-12345"), False,
                         "Wrong output for '-12345'")
        self.assertEqual(validate_pin("1.234"), False,
                         "Wrong output for '1.234'")
        self.assertEqual(validate_pin("00000000"), False,
                         "Wrong output for '00000000'")


if __name__ == "__main__":
    unittest.main()

File: JavaScript/7kyu/_123_Testing.js
import { strict as assert } from "assert";
import { getCount_2 } from "./VovewlCount.js";
import { accum } from "./Mumbling.js";
import { disemvowel } from "./DisemvowelTrolls.js";
import { highAndLow } from "./HighestAndLowest.js";
import { squareDigits } from "./Square Every Digit.js";
import { descendingOrder } from "./Descending Order.js";
import { getMiddle } from "./GetTheMiddleCharacter.js";
import { isIsogram } from "./Isograms.js";
import { filter_list } from "./ListFiltering.js";
import { XO } from "./ExesAndOhs.js";

describe("7kyu Tests", () => {
  it("Vovewl Count", function () {
    assert.strictEqual(getCount_2("abracadabra"), 5);
    assert.strictEqual(getCount_2("aeiou"), 5);
    assert.strictEqual(getCount_2("y"), 0);
    assert.strictEqual(getCount_2("bcdfghjklmnpqrstvwxz y"), 0);
    assert.strictEqual(getCount_2(""), 0);
  });
  it("Mumbling", () => {
    assert.strictEqual(
      accum("ZpglnRxqenU"),
      "Z-Pp-Ggg-Llll-Nnnnn-Rrrrrr-Xxxxxxx-Qqqqqqqq-Eeeeeeeee-Nnnnnnnnnn-Uuuuuuuuuuu"
    );
    assert.strictEqual(
      accum("NyffsGeyylB"),
      "N-Yy-Fff-Ffff-Sssss-Gggggg-Eeeeeee-Yyyyyyyy-Yyyyyyyyy-Llllllllll-Bbbbbbbbbbb"
    );
    assert.strictEqual(
      accum("MjtkuBovqrU"),
      "M-Jj-Ttt-Kkkk-Uuuuu-Bbbbbb-Ooooooo-Vvvvvvvv-Qqqqqqqqq-Rrrrrrrrrr-Uuuuuuuuuuu"
    );
    assert.strictEqual(
      accum("EvidjUnokmM"),
      "E-Vv-Iii-Dddd-Jjjjj-Uuuuuu-Nnnnnnn-Oooooooo-Kkkkkkkkk-Mmmmmmmmmm-Mmmmmmmmmmm"
    );
    assert.strictEqual(
      accum("HbideVbxncC"),
      "H-Bb-Iii-Dddd-Eeeee-Vvvvvv-Bbbbbbb-Xxxxxxxx-Nnnnnnnnn-Cccccccccc-Ccccccccccc"
    );
  });
  it("Disemvowel Trolls", () => {
    assert.strictEqual(
      disemvowel("This website is for losers LOL!"),
      "Ths wbst s fr lsrs LL!"
    );
    assert.strictEqual(
      disemvowel(
        "No offense but,\nYour writing is among the worst I've ever read"
      ),
      "N ffns bt,\nYr wrtng s mng th wrst 'v vr rd"
    );
    assert.strictEqual(
      disemvowel("What are you, a communist?"),
      "Wht r y,  cmmnst?"
    );
  });
  it("Highest and Lowest", () => {
    assert.strictEqual(
      highAndLow("4 5 29 54 4 0 -214 542 -64 1 -3 6 -6"),
      "542 -214"
    );
  });
  it("Square Every Digit", () => {
    assert.strictEqual(squareDigits(3212), 9414);
    assert.strictEqual(squareDigits(2112), 4114);
    assert.strictEqual(squareDigits(0), 0);
  });
  it("Descending Order", () => {
    assert.strictEqual(descendingOrder(0), 0);
    assert.strictEqual(descendingOrder(1), 1);
    assert.strictEqual(descendingOrder(111), 111);
    assert.strictEqual(descendingOrder(15), 51);
    assert.strictEqual(descendingOrder(1021), 2110);
    assert.strictEqual(descendingOrder(123456789), 987654321);
  });
  it("Get the middle character", function () {
    assert.strictEqual(getMiddle("test"), "es");
    assert.strictEqual(getMiddle("testing"), "t");
    assert.strictEqual(getMiddle("middle"), "dd");
    assert.strictEqual(getMiddle("A"), "A");
  });
  it("Isogram tests", () => {
    assert.strictEqual(isIsogram("Dermatoglyphics"), true);
    assert.strictEqual(isIsogram("isogram"), true);
    assert.strictEqual(
      isIsogram("aba"),
      false,
      "same chars may not be adjacent"
    );
    assert.strictEqual(
      isIsogram("moOse"),
      false,
      "same chars may not be same case"
    );
    assert.strictEqual(isIsogram("isIsogram"), false);
    assert.strictEqual(
      isIsogram(""),
      true,
      "an empty string is a valid isogram"
    );
  });
  it("List Filtering tests", () => {
    assert.deepEqual(filter_list([1, 2, "a", "b"]), [1, 2]);
    assert.deepEqual(filter_list([1, "a", "b", 0, 15]), [1, 0, 15]);
    assert.deepEqual(filter_list([1, 2, "aasf", "1", "123", 123]), [1, 2, 123]);
  });
  it("Exes and Ohs", () => {
    assert.strictEqual(XO("xo"), true);
    assert.strictEqual(XO("xxOo"), true);
    assert.strictEqual(XO("xxxm"), false);
    assert.strictEqual(XO("Oo"), false);
    assert.strictEqual(XO("ooom"), false);
    assert.strictEqual(XO("Oox"), false);
  });
});
